### ISSUE TITLE
`jsx-curly-brace-presence` conflicts with `no-unescaped-entities`

### ISSUE THREAD

**Issue by ntkoopman** [yannickcr/eslint-plugin-react#3214]
With both `jsx-curly-brace-presence` and `no-unescaped-entities` turned on
```
<script>{`window.foo = "bar"`}</script>
```
is fixed into
```
<script>window.foo = "bar"</script>
```
which works fine, but will give a warning/error on `no-unescaped-entities`. 

It is my understanding that `jsx-curly-brace-presence` should not be triggering on this code, since the documentation says:
> If the rule is set to get rid of unnecessary curly braces(JSX expression) and there are characters that need to be escaped in its JSX form, such as quote characters, [forbidden JSX text characters](https://facebook.github.io/jsx/), escaped characters and anything that looks like HTML entity names, the code will not be warned because the fix may make the code less readable.

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-curly-brace-presence.md

**Comment by aprilmintacpineda** [comment]
This is still happening though.

![image](https://user-images.githubusercontent.com/21032419/218438304-98e72416-9b86-4569-a0e4-67d10a4e8793.png)

After fixing the jsx-curly-brace-presence error

![image](https://user-images.githubusercontent.com/21032419/218438462-9309ab03-e308-465a-8010-6b6319c02c9f.png)

**Comment by ljharb** [comment]
@aprilmintacpineda can you be sure you’re on the latest version, and file a new issue if it’s still a problem?

**Comment by musjj** [comment]
It looks like that the special casing only considers backticks but not normal quotes.

This triggers a lint error:

```tsx
<p>{"'"}</p>
```

But this doesn't:

```tsx
<p>{`'`}</p>
```

**Comment by ljharb** [comment]
@musjj thanks, can you file a new issue for that one?

**Comment by musjj** [comment]
Made a new issue: https://github.com/jsx-eslint/eslint-plugin-react/issues/3801
